src/DynamoCore/Models/HomeWorkspaceModel.cs

```diff
@@ -20,6 +20,13 @@ public class HomeWorkspaceModel : WorkspaceModel
 
         private bool graphExecuted;              
         public readonly bool VerboseLogging;
+       
+        /// <summary>
+        /// This is set from Dynamoviewmodel. This is required here otherwise 
+        /// preview graph will run under every context that affects some of the test.
+        /// Also preview graph should run only when showrunpreview is set to true.
+        /// </summary>        
+        public bool ShowRunPreview { get; set; }
```

Comment:
Are we sure we want this to be part of the public API for HomeWorkspaceModel? Is there another way that we communicate this change from the UI without having to expose a public property? What would a 3rd party developer who is working with HomeWorkspaceModel think this does? 

Also, the comments on the property are the documentation for this level of the architecture. The model should not know that the view model exists, nor should its documentation :) This documentation needs to explain how to use this API as if the view model were non-existent.